h2o-core/src/main/java/water/util/FrameUtils.java

```diff
@@ -45,24 +46,40 @@ public static Frame parseFrame(Key okey, File ...files) throws IOException {
   /** Parse given set of URIs and produce a frame's key representing output.
    *
    * @param okey key for ouput frame. Can be null
-   * @param uris array of URI (file://, hdfs://, s3n://, s3a://, s3://, ...) to parse
+   * @param uris array of URI (file://, hdfs://, s3n://, s3a://, s3://, http://, https:// ...) to parse
    * @return a frame which is saved into DKV under okey
    * @throws IOException in case of parse error.
    */
   public static Frame parseFrame(Key okey, URI ...uris) throws IOException {
     return parseFrame(okey, null, uris);
   }
 
+  public static Key eagerLoadFromHTTP(String path) throws IOException {
+    java.net.URL url = new URL(path);
+    Key destination_key = Key.make(path);
+    java.io.InputStream is = url.openStream();
+    UploadFileVec.ReadPutStats stats = new UploadFileVec.ReadPutStats();
+    UploadFileVec.readPut(destination_key, is, stats);
+    return destination_key;
+  }
+
+
   public static Frame parseFrame(Key okey, ParseSetup parseSetup, URI ...uris) throws IOException {
     if (uris == null || uris.length == 0) {
       throw new IllegalArgumentException("List of uris is empty!");
     }
     if(okey == null) okey = Key.make(uris[0].toString());
     Key[] inKeys = new Key[uris.length];
-    for (int i=0; i<uris.length; i++)  inKeys[i] = H2O.getPM().anyURIToKey(uris[i]);
+    for (int i = 0; i < uris.length; i++){
```

Comment:
This is here to solve the Sparkling Water specific case, with ```new H2OFrame(URI..)```. When the user calls this, we also load data eagerly right away